src/TimeSeries/encodings/timeseriespreprocessing.jl

```diff
@@ -0,0 +1,77 @@
+struct TSStats{T}
+    means::AbstractArray{T,2}
+    stds::AbstractArray{T,2}
+end
+
+function TSStats(means, stds)
+    TSStats{eltype(means)}(means, stds)
+end
+
+"""
+    TSPreprocessing() <: Encoding
+
+Encodes 'TimeSeriesRow's by normalizing the time-series values. The time-series can
+either be normalized by each variable or time-step.
+
+Encodes
+- 'TimeSeriesRow' -> 'TimeSeriesRow'
+"""
+
+struct TSPreprocessing <: Encoding
+    tfms
+    stats::TSStats
+end
+
+function TSPreprocessing(stats::TSStats)
+    base_tfms = [
+        TSStandardize
+    ]
+    return TSPreprocessing{}(base_tfms, stats)
+end
+
+function TSStandardize(
+    obs,
+    p::TSPreprocessing
+)
+    means = p.stats.means
+    stds  = p.stats.stds
+    obs = obs .- means
+    obs = obs ./ stds
+    return obs
```

Comment:
```suggestion
    return (obs .- p.stats.means) ./ p.stats.stds
```
This will fuse and save an intermediate allocation.